src/main/java/com/theelixrlabs/healthcare/service/DoctorService.java

```diff
@@ -42,8 +42,8 @@ public DoctorService(DoctorRepository doctorRepository, DoctorPatientAssignmentR
      * @return true if the doctor is assigned to at least one patient, false otherwise.
      */
     private boolean isDoctorAssignedToPatient(UUID validDoctorId) {
-        Optional<DoctorPatientAssignmentModel> doctorAssignmentList = doctorPatientAssignmentRepository.findByDoctorIdAndDateOfUnassignmentNull(validDoctorId);
-        return doctorAssignmentList.isPresent();
+        List<DoctorPatientAssignmentModel> doctorAssignmentList = doctorPatientAssignmentRepository.findByDoctorIdAndDateOfUnassignmentNull(validDoctorId);
+        return !doctorAssignmentList.isEmpty();
```

Comment:
can you try using existsByDoctorIdAndDateOfUnassignmentNull(validDoctorId) ?

Reply:
changed to existsByDoctorIdAndDateOfUnassignmentNull(validDoctorId).